core-metastore/src/main/java/org/zalando/nakadi/repository/db/EventTypeRepository.java

```diff
@@ -102,6 +104,17 @@ public EventType mapRow(final ResultSet rs, final int rowNum) throws SQLExceptio
                 new EventTypeMapper());
     }
 
+    public List<EventType> list(final AuthorizationAttribute writer) {
+        return jdbcTemplate.query(
+                "SELECT et_event_type_object " +
+                        "FROM zn_data.event_type," +
+                        "jsonb_to_recordset(et_event_type_object->'authorization'->'writers') " +
+                        "AS writers(data_type text, value text) " +
+                        "WHERE writers.data_type = ? AND writers.value = ?",
+                Set.of(writer.getDataType(), writer.getValue()).toArray(),
```

Comment:
Is the order well-defied here as we pull from the set to an array?..

Reply:
Very good point